core/src/main/scala/chisel3/Mem.scala

```diff
@@ -60,42 +60,52 @@ sealed abstract class MemBase[T <: Data](val t: T, val length: BigInt) extends H
     */
   def apply(x: Int): T = macro SourceInfoTransform.xArg
 
+  def do_apply_impl(idx: UInt, clock: Clock)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T = {
+    if (clock != clockInst) {
+      Builder.warning("The clock used to initialize the memory is different than the one used to initialize the port. This behavior will be depracated in 3.6.0")
```

Comment:
```suggestion
      Builder.warning("The clock used to initialize the memory is different than the one used to initialize the port. This behavior will be deprecated in 3.6.0")
```

Don't we have a Chisel runtime deprecation we can use here?